### PR TITLE
Remove all server-side mustache code from zipkin-web

### PR DESCRIPTION
**Pull request by eirslett**
The server now only delivers a static index.html page,
which in turn loads JavaScript/CSS to run the application.
- Remove mustache dependencies from Gradle
- Add index.html "container page" for hosting the application
- Add new Flight UI component: "common", for rendering the
  application layout
- The EnvironmentUI component is now rendered by the "common"
  component
- Use events to trigger loading of serviceName/spanName to
  populate dropdowns

### CODE REVIEW

File: zipkin-web/src/main/resources/templates/v2/index.mustache [hunk 1]
```diff
@@ -2,19 +2,11 @@
   <form class="form-inline" role="form">
     <div class="form-group">
       <select data-placeholder="Service Name" class="form-control input-sm" style="width: 234px;" name="serviceName" id="serviceName">
-        <option value=""></option>
```
Comment (yurishkuro) [inline zipkin-web/src/main/resources/templates/v2/index.mustache:5]: in our installation, populating service names takes from one to several seconds.  Adrian tried to add caching control some time ago, but it doesn't look like it helped in our case. Does it makes sense to cache the whole HTML of the dropdown in the UI?

Comment (codefromthecrypt) [inline zipkin-web/src/main/resources/templates/v2/index.mustache:5]: Ps happy to dig deeper into the cache-control header thing if you want to
open an issue on it (or reopen the old one).

Comment (eirslett) [inline zipkin-web/src/main/resources/templates/v2/index.mustache:5]: It's all rendered based on JSON API calls, so we should rather add cache-control headers to the API. (In zipkin-query, and zipkin-web should forward the cache headers in CopyRenderer)

Comment (yurishkuro) [inline zipkin-web/src/main/resources/templates/v2/index.mustache:5]: mine concern is that retrieving 100s of service names even from cache and rebuilding the DOM is expensive.

Comment (eirslett) [inline zipkin-web/src/main/resources/templates/v2/index.mustache:5]: What is the expensive part of it? The call between the query server and the storage engine/cassandra or the browser rendering? It sounds strange that it takes such a long time... is it due to very many service names?
JavaScript converting a list of names into html is very quick (as quick as if it were done in Java) so it sounds like either IO or an expensive algorithm is to blame...

Does it take several seconds before this commit? Or only after this commit? Or regardless?

Comment (codefromthecrypt) [inline zipkin-web/src/main/resources/templates/v2/index.mustache:5]: This thread is about #718, so let's keep it contained there. There's been overwhelming support to undo the server-side mustache templates and I don't want to halt this PR on account of an old issue

**Pull request merged**